src/hooks/useTransactionNonces.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { Codec } from '@polkadot/types/types';
+
+import { useEffect, useState } from 'react';
+
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import useDashboard from './useDashboard';
+import useLaneId from './useLaneId';
+import useLoadingApi from './useLoadingApi';
+import useChainGetters from './useChainGetters';
+import { isTransactionCompleted } from '../util/transactionUtils';
+import { getSourceTargetRole } from '../util/chainsUtils';
+import { TransactionStatusType } from '../types/transactionTypes';
+import getSubstrateDynamicNames from '../util/getSubstrateDynamicNames';
+interface Props {
+  transaction: TransactionStatusType;
+}
+
+const useTransactionNonces = ({ transaction }: Props) => {
+  const [nonceOfTargetFinalizedBlock, setNonceOfTargetFinalizedBlock] = useState<null | number>(null);
+  const [latestReceivedNonceRuntimeApi, setLatestReceivedNonceRuntimeApi] = useState(0);
```

Comment:
This is set in async context so it should be using `isMounted()`.

Reply:
https://github.com/paritytech/parity-bridges-ui/pull/140/commits/b5fb972630cfbf19197e0823f6411cf641d30488